docs/src/content/docs/getting-started/setup.md

```diff
@@ -0,0 +1,218 @@
+})
+```
+
+Generics supported as well
+
+```ts
+/* tsconfig: strictNullChecks: true */
+// AtomMut<string | null>
+const nullableAtom = atom<string | null>(null) 
+```
+
+You can play with this example on [typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgQwMY2BAdgGhTCEOAXzgDMpC4ByAASgFNkCQB6VaB6gKF46wDO8LAFcQAQRZwAvPkIAKAMwBKANx9sQuEKiSqs5gupkIEamo2D4GMHqIGW8+egAeymQD5E3OHH5bRez8YFwA6ATAAT3lAu2UfP014HRlgsIjonTiEhOAyOBixOC8ABncEBN9GGBEoLDhA9V9SBgAbAQZvXyqGGrrtGCgE4m5iC25-YRFW1uQAI1aGO1TDEAAeHWAsAHM4AB8G6daPQpm1IA)
+
+### With ESlint
+
+We recommend using the `@reatom/eslint-plugin`.  
+This is optional, but greatly improves the development experience.
+
+#### Installation
+
+```
+npm i -D @reatom/eslint-plugin
+```
+
+
+#### Usage
+
+You should add @reatom to plugins and specify extends or rules into your config.
+
+```json
+{
+  "plugins": ["@reatom"],
+  "extends": ["plugin:@reatom/recommended"]
+}
+```
+
+#### Configuration
+
+Example of customizing rules:
+
+```json
+{
+  "plugins": ["@reatom"],
+  "rules": {
+    "@reatom/atom-rule": "error",
+    "@reatom/action-rule": "error",
+    "@reatom/reatom-prefix-rule": "error",
+    "@reatom/atom-postfix-rule": "error"
+  }
+}
+```
+
+More examples you can found in [@reatom/eslint-plugin](/package/eslint-plugin/) package documentation
+
+### With React
+
+#### Installation
+
+```
+npm i @reatom/npm-react
+```
+
+Then you need add add reatom wrapper
+
+React >=18:
+```jsx
+
+import { createCtx } from '@reatom/core'
+import { reatomContext } from '@reatom/npm-react'
+import { Main } from './path/to/an/Main';
+
+const ctx = createCtx()
+
+export const App = () => (
+  <reatomContext.Provider value={ctx}>
+    <Main />
+  </reatomContext.Provider>
+)
+
+```
+
+React 16 and 17:
+
+For `react-dom`:
+
+```js
+import { unstable_batchedUpdates } from 'react-dom'
+import { createCtx } from '@reatom/core'
+import { setupBatch, withBatching } from '@reatom/npm-react'
+import { Main } from './path/to/an/Main';
+
+setupBatch(unstable_batchedUpdates)
+const ctx = withBatching(createCtx())
+
+export const App = () => (
+  <reatomContext.Provider value={ctx}>
+    <Main />
+  </reatomContext.Provider>
+)
+
+```
+
+For `react-native`:
+
+```js
+import { unstable_batchedUpdates } from 'react-native'
+import { createCtx } from '@reatom/core'
+import { setupBatch } from '@reatom/npm-react'
+import { Main } from './path/to/an/Main';
+
+setupBatch(unstable_batchedUpdates)
+const ctx = withBatching(createCtx())
+
+export const App = () => (
+  <reatomContext.Provider value={ctx}>
+    <Main />
+  </reatomContext.Provider>
+)
+
+```
+
+#### Usage
+`useAtom` allow use atoms inside react components, and  
+`useAction` same but for actions.
+Here is how:
+
+```jsx
+const nameAtom = atom('Joe')
+const handleNameChange = action(
+  (ctx, event) => nameAtom(ctx, event.currentTarget.value),
+  'handleNameChange',
+)
+
+const Greeting = () => {
+  const [name] = useAtom(nameAtom)
```

Comment:
may be we will use `setName` here and don't mention `action` at all?

- less complexity
- `useAtom` will be more familiar and understandable with the setter, just like `useState`